### PR TITLE
Update version and make extra call to get user info for all user types

### PR DESCRIPTION
**Pull request by timomitchel**
We need to add a version to the unversioned `/me` endpoint Clever is deprecating at the end of June.

Here is the [canary](https://app.datadoghq.com/logs?query=empire.app.release%3Att-canary-oauth-clever-gem%20&cols=host%2Cservice&index=%2A&messageDisplay=inline&sort=time&stream_sort=desc&viz=stream&from_ts=1686770958439&to_ts=1686771858439&live=true) in prod. 

I tested this on students and teachers with and without identities prior to me logging in through the Clever portal.

### CODE REVIEW

File: lib/omniauth/strategies/clever.rb [hunk 1]
```diff
@@ -45,17 +45,73 @@ def callback_phase
...
-        { :user_type => raw_info['type'] }.merge! raw_info['data']
+        { :user_type => user_info['type'] }.merge! raw_info['data']
       end
 
       extra do
         {
           'raw_info' => raw_info
         }
       end
+      #  {
```
Comment (jt-traub) [inline lib/omniauth/strategies/clever.rb:58]: I would put a newline above this comment block to more tightly associate it with this function and a comment explaining that this block is an example of the data shape returned.
I would similarly add a comment to the other call and it's data shape explaining that.

**Review by jt-traub — APPROVED** [review]

File: lib/omniauth/strategies/clever.rb [hunk 2]
```diff
@@ -45,17 +45,73 @@ def callback_phase
...
+      #Should return "/v2.0/teachers/5798dc36fd5e25eb150010b5" from example above
+      def extract_canonical_uri
+         user_info['links'].find { |link| link['rel'] == 'canonical' }['uri']
+      end
 
+      # {
+      #   "data": {
+      #     "created": "2021-08-11T04:51:02.722Z",
+      #     "district": "573f84ce403fe40100000fb7",
+      #     "email": "carrie.davros@jeffco.k12.co.us",
```
Comment (sdawson) [inline lib/omniauth/strategies/clever.rb]: is this fake data? It just reads disconcertingly like a legit set of PII user info

Comment (timomitchel) [reply lib/omniauth/strategies/clever.rb]: Yea it absolutely was. I had put it there as a reminder of the object and then never went back to edit. Thank you my bad. Updated. @sdawson

File: lib/omniauth/strategies/clever.rb (hunk 1 above)
Comment (timomitchel) [reply lib/omniauth/strategies/clever.rb:58]: Done and done. Good call out.

**Review by jt-traub — APPROVED** [review]

**Pull request merged**